### PR TITLE
docs: ensure consistent casing of section headers

### PR DESCRIPTION
**Pull request by DamianGlowala**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Knock knock, case police, let's make the case consistent, PLEASE! ;)

- General inspiration for introducing consistency across the headers came from the Vue.js documentation.
- Whenever in doubt, I followed the Wikipedia Manual of Style available as one of the options at https://capitalizemytitle.com/style/Wiki.
- PR also includes other minor stylistic tweaks.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by danielroe** [comment]
Love it! Consistency is important. Though my own personal preference is not to use title case but only capitalise first word - wdyt @pi0 @Atinux @antfu?

**Comment by pi0** [comment]
I personally always capitalize titles to be more readable (and used this in couple of places) but totally fine to keep first start case

**Comment by antfu** [comment]
I don't have a strong opinion either. But This PR looks good to me as I suppose the title case is consistent and easy to follow.

**Review by Atinux — APPROVED** [review]

**Comment by Atinux** [comment]
It looks good to me to have this consistency for headings 😊

File: docs/content/2.guide/6.going-further/3.modules.md [hunk 1]
```diff
@@ -153,16 +153,16 @@ Consider documenting module usage in the readme file:
 
 Linking to the integration website and documentation is always a good idea.
 
-### Use `nuxt-` prefix for npm packages
+### Use `nuxt-` Prefix for npm Packages
```
Comment (pi0) [inline docs/content/2.guide/6.going-further/3.modules.md:156]: It is inconsistent though in some places like for `for npm` but in some other places we have `Features To Enable`.

**Comment by pi0** [comment]
What if we make an exception for non-keywords such as "to", "for", "with", etc to be lower cases?

File: docs/content/2.guide/6.going-further/3.modules.md (hunk 1 above)
Comment (danielroe) [reply docs/content/2.guide/6.going-further/3.modules.md:156]: Yes. Title case is somewhat tricky. `Features to Enable` is correct, if we are using it. https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case is helpful.

Comment (pi0) [reply docs/content/2.guide/6.going-further/3.modules.md:156]: /cc @antfu Is this something we can introduce in case-police?

**Comment by antfu** [comment]
> What if we make an exception for non-keywords such as "to", "for", "with", etc to be lower cases?

From my understanding, conjunctions should be lowercase already in the title.

File: docs/content/2.guide/6.going-further/3.modules.md (hunk 1 above)
Comment (pi0) [reply docs/content/2.guide/6.going-further/3.modules.md:156]: (Uptate: working on a local script to automate this)

Comment (antfu) [reply docs/content/2.guide/6.going-further/3.modules.md:156]: Case-police only checks brand/tools names, usually only one word. While for this case `npm` is supported to be all lowercase by their branding guide.

File: docs/content/2.guide/6.going-further/7.testing.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ The amount of time (in milliseconds) to allow for `setupTest` to complete its wo
 * Type: `number`
 * Default: `60000`
 
-### Features to enable
+### Features To Enable
```
Comment (pi0) [inline docs/content/2.guide/6.going-further/7.testing.md:71]:
```suggestion
### Features to Enable
```

**Review by pi0 — APPROVED** [review]

File: docs/content/2.guide/6.going-further/3.modules.md (hunk 1 above)
Comment (pi0) [reply docs/content/2.guide/6.going-further/3.modules.md:156]: ~> https://github.com/nuxt/framework/issues/6589

**Comment by pi0** [comment]
Thanks, @DamianGlowala <3 I've done a check and all titles seem now valid with [APA Style](https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case) (using [this tool](https://capitalizemytitle.com). It would be nice if we could make this process automated to avoid inconsistencies in the future (tracker: https://github.com/nuxt/framework/issues/6589)

**Pull request merged**